model/pipeline/src/main/resources/org/apache/beam/model/common_urns.md

```diff
@@ -0,0 +1,122 @@
+<!--
+
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# Apache Beam URNs
+
+This file serves as a central place to enumerate and document the various
+URNs used in the Beam portability APIs.
+
+
+## Core Transforms
+
+### org.apache.beam:transform:pardo:v1
+
+Represents Beam's parallel do operation.
+
+Payload: A serialized ParDoPayload proto.
+
+### org.apache.beam:transform:group_by_key:v1
+
+Represents Beam's group-by-key operation.
+
+Payload: None
+
+### org.apache.beam:transform:window_into:v1
+
+Payload: A windowing strategy id.
+
+### org.apache.beam:transform:flatten:v1
+
+### org.apache.beam:transform:read:v1
+
+
+## Combining
+
+If any of the combine operations are produced by an SDK, it is assumed that
+the SDK understands the last three combine helper operations.
+
+### org.apache.beam:transform:combine_globally:v1
+
+### org.apache.beam:transform:combine_per_key:v1
+
+### org.apache.beam:transform:combine_grouped_values:v1
+
+### org.apache.beam:transform:combine_pgbkcv:v1
+
+### org.apache.beam:transform:combine_merge_accumulators:v1
+
+### org.apache.beam:transform:combine_extract_outputs:v1
+
+
+## Other common transforms
+
+### org.apache.beam:transform:reshuffle:v1
+
+
+## WindowFns
+
+### org.apache.beam:windowfn:global_windows:v1
+
+### org.apache.beam:windowfn:fixed_windows:v1
+
+### org.apache.beam:windowfn:sliding_windows:v1
+
+### org.apache.beam:windowfn:session_windows:v1
+
+
+## Coders
+
+###  org.apache.beam:coder:bytes:v1
+
+Components: None
+
+###  org.apache.beam:coder:varint:v1
+
+Components: None
+
+###  org.apache.beam:coder:kv:v1
+
+Components: The key and value coder, in that order.
+
+###  org.apache.beam:coder:iterable:v1
+
+Encodes an iterable of elements.
+
+Components: Coder for a single element.
+
+## Internal coders
+
+The following coders are typically not specified by manually by the user,
+but are used at runtime and must be supported by every SDK.
+
+###  org.apache.beam:coder:length_prefix:v1
+
+###  org.apache.beam:coder:global_window:v1
+
+###  org.apache.beam:coder:interval_window:v1
+
+###  org.apache.beam:coder:windowed_value:v1
+
+
+## Side input access
+
+### org.apache.beam:side_input:iterable:v1
```

Comment:
This one shouldn't exist/be removed.

Reply:
I was looking at what it would take to remove this from the Python SDK, and given that the existing batch code doesn't support indexable side inputs I think we should offer this option as well. (This seems a better option than generating different graphs when trying to run portably.)

Reply:
How about we drop it for now and just use it inside the Python SDK till there is a need. I don't want this to make its way out of the Python SDK till there is a strong need for iterable side inputs over the Fn API.